File: examples/metal/reserved_ip_block/example_2/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/metal"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		example, err := metal.NewReservedIpBlock(ctx, "example", &metal.ReservedIpBlockArgs{
			ProjectId: pulumi.Any(projectId),
			Metro:     pulumi.String("sv"),
			Quantity:  pulumi.Int(2),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewDevice(ctx, "nodes", &metal.DeviceArgs{
			ProjectId:       pulumi.Any(projectId),
			Metro:           pulumi.String("sv"),
			Plan:            pulumi.String(metal.PlanC3SmallX86),
			OperatingSystem: pulumi.String(metal.OperatingSystem_Ubuntu20_04),
			Hostname:        pulumi.String("test"),
			BillingCycle:    pulumi.String(metal.BillingCycleHourly),
			IpAddresses: metal.DeviceIpAddressArray{
				&metal.DeviceIpAddressArgs{
					Type: pulumi.String("public_ipv4"),
					Cidr: pulumi.Int(31),
					ReservationIds: pulumi.StringArray{
						example.ID(),
					},
				},
				&metal.DeviceIpAddressArgs{
					Type: pulumi.String("private_ipv4"),
				},
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: examples/metal/reserved_ip_block/example_2/python/__main__.py
import pulumi
import pulumi_equinix as equinix

example = equinix.metal.ReservedIpBlock("example",
    project_id=project_id,
    metro="sv",
    quantity=2)
nodes = equinix.metal.Device("nodes",
    project_id=project_id,
    metro="sv",
    plan=equinix.metal.Plan.C3_SMALL_X86,
    operating_system=equinix.metal.OperatingSystem.UBUNTU20_04,
    hostname="test",
    billing_cycle=equinix.metal.BillingCycle.HOURLY,
    ip_addresses=[
        {
            "type": "public_ipv4",
            "cidr": 31,
            "reservation_ids": [example.id],
        },
        {
            "type": "private_ipv4",
        },
    ])
